scripts/SANS/sans/user_file/parser_helpers/wavelength_parser.py

```diff
@@ -0,0 +1,84 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from typing import Tuple, List
+
+from mantid.py36compat import dataclass
+from sans.common.enums import RangeStepType
+from sans.common.general_functions import get_ranges_from_event_slice_setting
+from sans.state.StateObjects.StateCalculateTransmission import StateCalculateTransmission
+from sans.state.StateObjects.StateNormalizeToMonitor import StateNormalizeToMonitor
+from sans.state.StateObjects.StateWavelength import StateWavelength
+from sans.state.StateObjects.StateWavelengthAndPixelAdjustment import StateWavelengthAndPixelAdjustment
+from sans.user_file.parser_helpers.toml_parser_impl_base import TomlParserImplBase
+
+
+@dataclass(init=True)
+class DuplicateWavelengthStates:
+    """
+    These four classes contain duplicated attributes, so this POD
+    class ties them together for subsequent setters
+    """
+    transmission: StateCalculateTransmission
+    normalize: StateNormalizeToMonitor
+    wavelength: StateWavelength
+    pixel: StateWavelengthAndPixelAdjustment
+
+    def iterate_fields(self):
+        return [self.transmission, self.normalize,
+                self.wavelength, self. pixel]
+
+
+def parse_range_wavelength(wavelength_range: str) -> Tuple[List, List]:
+    """
+    Parses a wavelength range and outputs a tuple of wavelength_start
+    and wavelength_stop in the format StateObjects expect.
+    This method exists to unify the GUI and TOML file parsing into a single place
+    return: ([wavelength_start], [wavelength_stop])
+    """
+    wavelength_start, wavelength_stop = get_ranges_from_event_slice_setting(wavelength_range)
+    wavelength_start = [min(wavelength_start)] + wavelength_start
```

Comment:
It's because the existing data structure (later changed in #29939 ) is spread over two lists. So 

(2,4), (4, 6), (6,8)

Turns into
```
wavelength_start = [2, 4, 6]
wavelength_stop = [4, 6, 8]
```

Then for added complexity someone decided to prepend the entire range to the start of the list, to make it easier for the algs. So we now have
```
wavelength_start = [2, 2, 4, 6]  # where 2 is the start
wavelength_stop = [8, 4, 6, 8]  # where 8 is the end
```

Needless to say, this was replaced in the aforementioned PR to
```
wavelength_full_range = (2, 8)
wavelength_intervals = [(2, 8), (2,4), (4, 6), (6, 8)]  # As we still need to process 2,8 like a user selected step
```